NitroxClient/MonoBehaviours/ThrottledBuilder.cs

```diff
@@ -109,206 +98,29 @@ private void ActionBuildEvent(BuildEvent buildEvent)
-                }
-
-                GameObject finishedPiece = null;
-
-                // There can be multiple objects in a cell (such as a corridor with hatces built into it)
-                // we look for a object that is able to be deconstucted that hasn't been tagged yet.
-                foreach (Transform child in cellTransform)
-                {
-                    bool isNewBasePiece = (child.GetComponent<NitroxEntity>() == null &&
-                                           child.GetComponent<BaseDeconstructable>() != null);
-
-                    if (isNewBasePiece)
-                    {
-                        finishedPiece = child.gameObject;
-                        break;
-                    }
-                }
-                
-                Validate.NotNull(finishedPiece, "Could not find finished piece in cell " + latestCell);
-
-                Log.Info("Construction completed on a base piece: " + constructionCompleted.PieceId + " " + finishedPiece.name);
-
-                UnityEngine.Object.Destroy(constructableBase.gameObject);
-                NitroxEntity.SetNewId(finishedPiece, constructionCompleted.PieceId);
-                
-                BasePieceSpawnProcessor customSpawnProcessor = BasePieceSpawnProcessor.From(finishedPiece.GetComponent<BaseDeconstructable>());
-                customSpawnProcessor.SpawnPostProcess(latestBase, latestCell, finishedPiece);
-            }
-            else
-            {
-                Constructable constructable = constructing.GetComponent<Constructable>();
-                constructable.constructedAmount = 1f;
-                constructable.SetState(true, true);
-
-                Log.Info("Construction completed on a piece of furniture: " + constructionCompleted.PieceId + " " + constructable.gameObject.name);
-            }
-            
-            if (constructionCompleted.BaseId != null && !NitroxEntity.GetObjectFrom(constructionCompleted.BaseId).HasValue)
-            {
-                Log.Info("Creating base: " + constructionCompleted.BaseId);
-                ConfigureNewlyConstructedBase(constructionCompleted.BaseId);
-            }
-        }
-
-        private void ConfigureNewlyConstructedBase(NitroxId newBaseId)
-        {
-            Optional<object> opNewlyCreatedBase = TransientLocalObjectManager.Get(TransientLocalObjectManager.TransientObjectType.BASE_GHOST_NEWLY_CONSTRUCTED_BASE_GAMEOBJECT);
-
-            if (opNewlyCreatedBase.HasValue)
-            {
-                GameObject newlyCreatedBase = (GameObject)opNewlyCreatedBase.Value;
-                NitroxEntity.SetNewId(newlyCreatedBase, newBaseId);
-            }
-            else
-            {
-                Log.Error("Could not assign new base id as no newly constructed base was found");
-            }
+            NitroxServiceLocator.LocateService<Building>().Constructable_ConstructionCompleted_Remote(constructionCompleted.PieceId);
```

Comment:
Let's use dependency injection through the constructor instead for all of these methods.

Reply:
What do you mean with depedeny injection?

Reply:
We only want to call the locator once - similar to the other items in the Start method above.

Reply:
Sorry i still don't understand what you are suggesting. The locator also gets called everytime an action is happening in a patch and a call is needed to a gamelogic class. I understand the locator as a dynamic reference to a static instance which has a permanent lifetime. I don't get the point why this would be harmful here to call it everytime a buildevent is processed.